earn/src/components/borrow/modal/RemoveCollateralModal.tsx

```diff
@@ -64,25 +70,34 @@ function RemoveCollateralButton(props: RemoveCollateralButtonProps) {
   const { activeChain } = useContext(ChainContext);
 
   const [isPending, setIsPending] = useState(false);
+  const [oracleSeed, setOracleSeed] = useChainDependentState<number | undefined>(undefined, activeChain.id);
+
+  const provider = useProvider({ chainId: activeChain.id });
 
   const isToken0Collateral = collateralToken.address === marginAccount.token0.address;
 
   const amount0 = isToken0Collateral ? collateralAmount : GN.zero(collateralToken.decimals);
   const amount1 = isToken0Collateral ? GN.zero(collateralToken.decimals) : collateralAmount;
 
+  useEffectOnce(() => {
+    (async () => {
+      const seed = await computeOracleSeed(marginAccount.uniswapPool, provider, activeChain.id);
+      setOracleSeed(seed);
+    })();
+  });
+
+  const borrowerInterface = useMemo(() => new ethers.utils.Interface(borrowerABI), []);
+  const encodedData = useMemo(
+    () => borrowerInterface.encodeFunctionData('transfer', [amount0.toBigNumber(), amount1.toBigNumber(), userAddress]),
+    [amount0, amount1, borrowerInterface, userAddress]
+  );
+
   const { config: removeCollateralConfig } = usePrepareContractWrite({
     address: marginAccount.address,
     abi: borrowerABI,
     functionName: 'modify',
-    args: [
-      ALOE_II_WITHDRAW_MANAGER_ADDRESS[activeChain.id],
-      ethers.utils.defaultAbiCoder.encode(
-        ['uint256', 'uint256', 'address'],
-        [amount0.toBigNumber(), amount1.toBigNumber(), userAddress]
-      ) as Address,
-      [isToken0Collateral, !isToken0Collateral],
-    ],
-    enabled: !!userAddress && collateralAmount.isGtZero() && collateralAmount.lte(userBalance),
+    args: [ALOE_II_SIMPLE_MANAGER_ADDRESS[activeChain.id], encodedData as Address, oracleSeed ?? 0],
```

Comment:
^^

Reply:
Also would prefer `encodedData` as 0xstring, not Address (it's not an Address). I know it's the same behind the scenes but still.